### PR TITLE
HDDS-9502. Migrate tests in ozonefs-common to JUnit5

### PR DESCRIPTION
**Pull request by VarshaRaviCV**
## What changes were proposed in this pull request?
HDDS-9502. Migrate tests in ozonefs-common to JUnit5

Migrating all the tests under ozonefs-common to Junit5 and making sure that Junit4 is not used in any of the tests. This PR also migrates the parametrized tests under ozonefs-common to junit5.
Tested that using grep -rlE "org.junit.[A-Z]" . --include '*.java' command in hadoop-ozone/ozonefs-common folder

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9502

## How was this patch tested?

This PR was tested by running the full CI - https://github.com/VarshaRaviCV/ozone/actions/runs/6942165040

### CODE REVIEW

**Comment by VarshaRaviCV** [comment]
@adoroszlai @Galsza Please review this PR

**Review by adoroszlai — COMMENTED** [review]
Thanks @VarshaRaviCV for the patch.

File: hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestBasicOzoneFileSystems.java [hunk 1]
```diff
@@ -35,63 +34,59 @@
...
-
-  private final FileSystem subject;
-
-  @Parameterized.Parameters
   public static Collection<Object[]> data() {
     return Arrays.asList(
         new Object[]{new BasicOzoneFileSystem()},
         new Object[]{new BasicRootedOzoneFileSystem()}
     );
   }
```
[the remark this answers is not in the document]
Comment (VarshaRaviCV) [reply hadoop-ozone/ozonefs-common/src/test/java/org/apache/hadoop/fs/ozone/TestBasicOzoneFileSystems.java]: Done. I have made the change.

File: hadoop-ozone/ozonefs-common/pom.xml
[the remark this answers is not in the document]
Comment (VarshaRaviCV) [reply hadoop-ozone/ozonefs-common/pom.xml:1]: Done.

**Review by adoroszlai — APPROVED** [review]
Thanks @VarshaRaviCV for updating the patch.

**Pull request merged**